app/uk/gov/hmrc/residencenilratebandcalculator/repositories/SessionRepository.scala

```diff
@@ -18,21 +18,61 @@ package uk.gov.hmrc.residencenilratebandcalculator.repositories
 
 import javax.inject.{Inject, Singleton}
 
-import play.api.libs.json.Json
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.{Configuration, Logger}
+import play.api.libs.json.{JsValue, Json}
 import play.modules.reactivemongo.MongoDbConnection
 import reactivemongo.api.DefaultDB
+import reactivemongo.api.indexes.{Index, IndexType}
 import reactivemongo.bson.{BSONDocument, BSONObjectID}
 import uk.gov.hmrc.http.cache.client.CacheMap
 import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class ReactiveMongoRepository(mongo: () => DefaultDB)
-  extends ReactiveRepository[CacheMap, BSONObjectID]("residenceNilRateBand", mongo, CacheMap.formats) {
+case class DatedCacheMap(id: String,
+                         data: Map[String, JsValue],
+                         createdAt: DateTime = DateTime.now(DateTimeZone.UTC))
```

Comment:
This field is behaving more like a "last updated" field as it is updated every time the record is updated - can we rename it to reflect that?